api/apps/api/src/utils/config.utils.spec.ts

```diff
@@ -8,6 +8,14 @@ process.env.NETWORK_CORS_ORIGINS = extraCorsOrigin;
 import { AppConfig } from './config.utils';
 
 describe('AppConfig', () => {
+  beforeAll(() => {
+    if (process.env.NODE_CONFIG_DIR !== 'apps/api/config') {
+      throw Error(
+        `Running the test suite with NODE_CONFIG_DIR=${process.env.NODE_CONFIG_DIR}, which may cause this test to fail. Please use NODE_CONFIG_DIR=apps/api/config.`,
+      );
+    }
```

Comment:
due to how `AppConfig` is implemented, even just importing it (which is pretty common in an indirect way, due to a lot of usage everywhere) makes it load the config from envs. Could be a good idea to move it to global setup for unit tests.

In the long run, I would suggest to replace the mechanism with https://docs.nestjs.com/techniques/configuration

Reply:
agree, the Nest-y way would be handy as part of a future tech debt cleanup (and alignment with the Nest way wherever there's no need to go ad-hoc).